tensorcircuit/templates/lattice.py

```diff
@@ -0,0 +1,1526 @@
+                        "linestyle": "-",
+                        "alpha": 0.6,
+                        "zorder": 1,
+                    }
+                    if bond_kwargs:
+                        plot_bond_kwargs.update(bond_kwargs)
+
+                    if self.dimensionality > 2:
+                        ax_3d = cast("Axes3D", ax)
+                        for i, j in bonds:
+                            p1, p2 = self._coordinates[i], self._coordinates[j]
+                            ax_3d.plot(
+                                [p1[0], p2[0]],
+                                [p1[1], p2[1]],
+                                [p1[2], p2[2]],
+                                **plot_bond_kwargs,
+                            )
+                    else:
+                        for i, j in bonds:
+                            p1, p2 = self._coordinates[i], self._coordinates[j]
+                            if self.dimensionality == 1:  #  type: ignore
+
+                                ax.plot([p1[0], p2[0]], [0, 0], **plot_bond_kwargs)  # type: ignore
+                            else:  # dimensionality == 2
+                                ax.plot([p1[0], p2[0]], [p1[1], p2[1]], **plot_bond_kwargs)  # type: ignore
+
+                except ValueError as e:
+                    logger.info(f"Could not draw bonds: {e}")
+
+        ax.set_title(f"{self.__class__.__name__} ({self.num_sites} sites)")
+        if self.dimensionality == 2:
+            ax.set_aspect("equal", adjustable="box")
+        ax.set_xlabel("x")
+        if self.dimensionality > 1:
+            ax.set_ylabel("y")
+        if self.dimensionality > 2 and hasattr(ax, "set_zlabel"):
+            ax.set_zlabel("z")
+        ax.grid(True)
+
+        # 3.  whether plt.show()
+        if fig_created_internally:
+            plt.show()
+
+    def _identify_distance_shells(
+        self,
+        all_distances_sq: Union[Coordinates, List[float]],
+        max_k: int,
+        tol: float = 1e-6,
+    ) -> List[float]:
+        """Identifies unique distance shells from a list of squared distances.
+
+        This helper function takes a flat list of squared distances, sorts them,
+        and identifies the first `max_k` unique distance shells based on a
+        numerical tolerance.
+
+        :param all_distances_sq: A list or array
+            of all squared distances between pairs of sites.
+        :type all_distances_sq: Union[np.ndarray, List[float]]
+        :param max_k: The maximum number of neighbor shells to identify.
+        :type max_k: int
+        :param tol: The numerical tolerance to consider two distances equal.
+        :type tol: float
+        :return: A sorted list of squared distances representing the shells.
+        :rtype: List[float]
+        """
+        if not isinstance(all_distances_sq, np.ndarray):
+            all_distances_sq = np.array(all_distances_sq)
+
+        sorted_dist_sq = np.sort(all_distances_sq[all_distances_sq > 1e-12])
```

Comment:
why not tol**2 but an explicit value, no need to change the code for this function?